src/filesystem.cpp

```diff
@@ -256,3 +256,33 @@ string shasta::filesystem::getAbsolutePath(const string& path)
     ::realpath(path.c_str(), buffer.data());
     return string(buffer.data());
 }
+
+
+string shasta::filesystem::executablePath() {
+    string path;
+    vector<char> buf(1024, 0);
```

Comment:
Consider changing 1024 to `PATH_MAX`, even though `PATH_MAX` is flawed in some ways?